src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackReader.cs

```diff
@@ -949,6 +949,7 @@ private static Exception ThrowInvalidCode(byte code)
         /// </summary>
         /// <param name="condition">A boolean value.</param>
         /// <exception cref="EndOfStreamException">Thrown if <paramref name="condition"/> is <see langword="false"/>.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
This is already a super-simple method that was designed to be naturally inlined (by avoiding a `throw` statement).
When I put it [into sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgEkB5AbhvqdYBKAVwB2GAJb4YLAMIR8AB3EAbGFADKagG7iwMXD2o1YAMwYLsUCdmUNcGKELAYGAWX25sAcxgAFbGAA1gIw2AAmajQA3jQMcQziYuYQuOISECIMhvEMsTn5TADMDBDAAFYwzgwh4QCiCBgwIqkZAGLQ+NgYABKhEVAAFACUeQUx1AUFAPRTDL5QEI1V3tiJ9gyQUI4KjWElUAz44rhgABbiMDoQQrgMYV3YDBinXYfYAJ4MquFPEAw2yggYC6iS8DAA7h9fn98E5TocYPhoO8WKNJnEACqnBbgtjNIQmEy6C5iABCBJMagAqiJVLhcAMFCk0uIMgwAHwAXgYhSGhjR6OIAHYGCIYODhtlJgBfGgC+IzADa7meEDCbEUygGKtOao1CmUHB2rOaLAAchA8cpEqChgBdeVxJU6vWa7UwVXqzVG9KmgCCXi8sHp4i0MCtNpEXntjvMUFDXRg/CQTBQDCxOLxuApxKaGHJhOptI8A2AEAgtkgIjCLIyIwmk3G6Jy4jMAwAhFWa776834k2+/kMxBwRaMLURNcvKdSe9Grh6npjRlJbGcrKGwUN7GZrMADzZ/CdKDvdk7mbp7Ej27YTIwBBL30Jau6EFRp4vFzPNRJyxNADkLhNFO8LAHO+gMLAnSRmCiQfkmwAUmovx2Bg0AwOeszfnG8jHDAez3Bg2Copu+S7gwe5TIex6nrGCjxloibJgwE5hBwJjqA4oT4IuMDLpkw6josE4gbO868fxwwMJy7Ifjioriix1bsZxsDYDxD58b6q5GKROTkQeQhHpYtF6QqF6CTed6afxz41sCEjvs8rzfrA/ywCIgEMMBQjTgwYHzpBiKrCIoLPvB/lIQcaGoehmERfRuG4PhdwPCR6IGVRRk0WeZlxPRCaNMxrEqVx6kSU+gljiJvkzuBC42dpFUms+YpQM1dbSbJzzyWK4JKWxHFlRpj4mgMfUDaVakjVpY1DCw7j0t4MBoK1agdSIfJytQ0pAA==) you can see how adding the attribute doesn't change the compiled assembly at all (yet adding NoInlining _does_ add a `call` to the assembly).
Are you actually seeing this in your traces as being called?